# Assignent 3
### Author: Ben Alpert
#### Collaborated with Jonathan Geis and Sofiya Elyukin, who previously worked with Sebastian Bana, Marc Toneatto. Code is taken from skeleton notebook provided in assignment instructions.

In [6]:
from __future__  import print_function, division
import sys
import os
import numpy as np

### Ever employed in a CEO Transitional Job
Null Hypothesis: The percent of former prisoners employed 3 years after release is the same or lower for candidates who participated in the program as for the control group. 

'P0' is the percentage of former prisoners employed who did participate in the program, and 'P1' is the percentage of former prisoners employed who did not participate in the program.

    H0: P0 - P1 <= 0
    H1: P0 - P1 > 0
    α = 0.05

In [14]:
alpha = 0.05
p0 = 70.1 * 0.01
p1 = 3.5 * 0.01
n0 = 564
n1 = 409

Nt0 = p0 * n0
Nt1 = p1 * n1

In [15]:
#define the sample proportion first
sp = (Nt0 + Nt1) / (n1 + n0)
print (sp)

0.421047276465


In [16]:
pHat = lambda p0, p1, n0, n1: (p0 * n0 + p1 * n1) / (n0 + n1)   #same as sample proportion
stdErr = lambda pHat, n0, n1: np.sqrt(pHat * (1 - pHat) * (1.0 / n0 + 1.0 / n1))

In [17]:
zscore = lambda p0, p1, s : (p0 - p1) / s
zEmploy = zscore(p0, p1, stdErr(pHat(p0, p1, n0, n1), n0, n1))

print ('The Z-score is: ', zEmploy)

The Z-score is:  20.7697865408


In [18]:
# Find the p-value using the probability for z-score 20.7698 taken from the z-score table.
pValEmploy = 1 - 0.9998

def report_result(p,a):
    print ('Is the p value ' + '{0:.4f} smaller than the critical value {1:.2f}?'.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('The Null hypothesis is {}'.format('rejected' if p < a  else 'not rejected') )

report_result(pValEmploy, alpha)

Is the p value 0.0002 smaller than the critical value 0.05?
YES!
The Null hypothesis is rejected


### Convicted of a Felony
Null Hypothesis: The percent of former prisoners convicted of a felony 3 years after release is the same or higher for candidates who participated in the program as for the control group. 

'P0' is the percentage of former prisoners convicted of a felony who did participate in the program, and 'P1' is the percentage of former prisoners convicted of a felony who did not participate in the program.
    
    H0: P0 - P1 >= 0
    H1: P0 - P1 < 0
    α = 0.05

In [20]:
alpha = 0.05
p0 = 10 * 0.01
p1 = 11.7 * 0.01
n0 = 568
n1 = 409

Nt0 = p0 * n0
Nt1 = p1 * n1

In [21]:
#define the sample proportion first
sp = (Nt0 + Nt1) / (n1 + n0)
print (sp)

0.107116683726


In [22]:
pHat = lambda p0, p1, n0, n1: (p0 * n0 + p1 * n1) / (n0 + n1)   #same as sample proportion
stdErr = lambda pHat, n0, n1: np.sqrt(pHat * (1 - pHat) * (1.0 / n0 + 1.0 / n1))

In [23]:
zscore = lambda p0, p1, s : (p0 - p1) / s
zRecid = zscore(p0, p1, stdErr(pHat(p0, p1, n0, n1), n0, n1))
print ('The z-score is: ', zRecid)

The z-score is:  -0.84764020522


In [24]:
# Find the p-value using the probability for z-score -0.848 taken from the z-score table.

pValRecid = 1 - 0.7995

def report_result(p,a):
    print ('Is the p value ' + '{0:.4f} smaller than the critical value {1:.2f}?'.format(p,a))
    if p < a:
        print ("YES!")
    else: 
        print ("NO!")
    
    print ('The Null hypothesis is {}'.format('rejected' if p < a  else 'not rejected') )

report_result(pValRecid, alpha)

Is the p value 0.2005 smaller than the critical value 0.05?
NO!
The Null hypothesis is not rejected


### Chi-squared

Create a contingency table for the values for the chi-squared test

|Convicted of a Felony      |        Yes             |        No         |          |    
|---------------------------|:----------------------:|------------------:|----------|
| Test Sample               | 0.100*568              | 0.900*568         | 568      |
| Control Sample            | 0.117*409              | 0.883*409         | 409      |
|                           |                        |                   |          |
| Total                     | 104.653                |  872.347          | 977      |

In [25]:
# from evalChisq.py: https://github.com/fedhere/PUI2016_fb55/blob/master/HW4_fb55/evalChisq.py

def evalChisq(values):
    values = np.array(values)
    E = np.empty_like(values)
    for j in range(len(values[0])):
        for i in range(2):
            
            E[i][j] = ((values[i,:].sum() * values[:,j].sum()) / 
                        (values).sum())
    return ((values - E)**2 / E).sum()

In [27]:
sample_values = np.array([[0.100 * 568, 0.900 * 568], [0.117 * 409, 0.883 * 409]])
print ('The Chi-sqaure value is: ', evalChisq(sample_values))

The Chi-sqaure value is:  0.718493917505


The minimum chi-square value is 3.84 based on DF = 1 and alpha = 0.05. Our schi-square value is 0.718, which is less than 3.84, therefore we cannot reject the null hypothesis at a significance level of alpha = 0.05.